Tools/TestWebKitAPI/Tests/WebKitCocoa/TextExtractionTests.mm

```diff
@@ -223,4 +225,36 @@ - (NSString *)debugDescriptionInWebView:(WKWebView *)webView error:(NSError **)o
     }
 }
 
+TEST(TextExtractionTests, TargetNode)
+{
+    RetainPtr webView = adoptNS([[TestWKWebView alloc] initWithFrame:CGRectMake(0, 0, 800, 600) configuration:^{
+        RetainPtr configuration = adoptNS([[WKWebViewConfiguration alloc] init]);
+        [[configuration preferences] _setTextExtractionEnabled:YES];
+        return configuration.autorelease();
+    }()]);
+
+    [webView synchronouslyLoadTestPageNamed:@"debug-text-extraction"];
+    [webView stringByEvaluatingJavaScript:@"getSelection().selectAllChildren(document.querySelector('h3[aria-label=\"Heading\"]'))"];
```

Comment:
do you need the string? and/or, you could also get away with `evaluateJavascript:`?

Reply:
I guess it's a little cumbersome to set that up since you'd have to wait for the completion handler to get called.